deis/builder
# fix(pkg/git/git.go): write the pre-receive hook script directly to the file descriptor

# Summary of Changes

Instead of creating the pre-receive hook script and writing the results to an intermediate `*bytes.Buffer`, this change writes the script's results directly to an `*os.File`.

# Issue(s) that this PR Closes

Fixes #321 

# Associated [End To End Test](https://github.com/deis/workflow-e2e) PR(s)

None. This path only changes an implementation detail.

# Associated [Documentation](https://github.com/deis/workflow) PR(s)

None. This path only changes an implementation detail.

# Associated [Design Document](http://docs.deis.io/en/latest/contributing/design-documents)(s)

None. This path only changes an implementation detail.

# Testing Instructions

1. Create a Deis Cluster
2. Register an app with `deis create`
3. Add your SSH key with `deis keys:add`
4. `git push deis master` from your Buildpack or Dockerfile app (for example, [example-go](https://github.com/deis/example-go))

The application should be successfully deployed.

# Pull Request Hygiene TODOs

Please make sure the below checklist is complete.
